### PR TITLE
fix(42519): 'Infer function return type' quick fix generates invalid code for arrow functions without parens

### LINKED ISSUE #42519: 'Infer function return type' quick fix generates invalid code for arrow functions without parens
# Bug Report

### 🔎 Search Terms
- Quick fix
- code action
- infer function return type

### 🕗 Version & Regression Information
TypeScript version: 4.2.0-dev.20210125

### 💻 Code
```ts
const a = async x => {
    return 1; 
}
```

Trigger the `infer function return type` fix here

### 🙁 Actual behavior

This results in:

```ts
const a = async x: Promise<number> => {
    return 1;
}
```

### 🙂 Expected behavior

Instead a set of parameter parens needs to be added in this case:

```ts
const a = async (x): Promise<number> => {
    return 1;
}
```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #42519

### CODE REVIEW

**Comment by a-tarasyuk** [comment]
@jessetrinity Could you please review this PR? It related to QF `Infer function return type`.

File: tests/cases/fourslash/refactorInferFunctionReturnType23.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/// <reference path='fourslash.ts' />
+
+////const foo = async /*a*//*b*/a => {
```
Comment (jessetrinity) [inline tests/cases/fourslash/refactorInferFunctionReturnType23.ts:3]: Is this a duplicate of `refactorInferFunctionReturnType23.ts`?

Comment (a-tarasyuk) [reply tests/cases/fourslash/refactorInferFunctionReturnType23.ts]: @jessetrinity Yes, you are right. It seems the conflicts were not resolved correctly with https://github.com/microsoft/TypeScript/pull/42737. Thanks for pointing that out. ;)

**Review by jessetrinity — APPROVED** [review]
Thanks!

**Pull request merged**